### ISSUE TITLE
Lint rule for non-bare specifier

### ISSUE THREAD

**Issue by dsherret** [denoland/deno_lint#1238]
Bad:

```ts
import * as path from "https://deno.land/std/path/mod.ts";
```

Good (uses import map):

```ts
import * as path from "@std/path";
```

Why: I want to ensure my dependenices are defined in the deno.json file instead of accidentally in the code (ex. via an auto-import going crazy) so that I can generate the lockfile for a project when running `deno install` (instead of also having to run/cache scripts with non-bare specifiers).
